frontend/src/components/auth/GroupMembers.vue

```diff
@@ -1,156 +1,162 @@
-          icon: "person_add",
-          text: "Add",
-          callback: this.addMembers
-        })
-      ]
-    }
+const props = defineProps({
+  group: {
+    required: false,
+    type: Object,
   },
-  watch: {
-    async group() {
-      await this.getMembers();
-    }
-  },
-  async activated() {
-    await this.getMembers();
-  },
-  methods: {
-    async onSelect() {
-      const elems = document.getElementsByName("select-users");
-      this.selected = Array.prototype.filter.call(elems, e => e.checked).map(e => e.value);
-    },
-    async addMembers() {
-      const boxes = document.getElementsByName("add-users");
-      const addUserIds = Array.from(boxes)
-          .filter(x => x.checked)
-          .map(x => x.value);
-      const response = this.$api.addMembers({groupId: this.group.id, userIds: addUserIds});
-      if (response === true) {
-        await this.getMembers();
-      }
-    },
-    async getMembers() {
-      this.loading = true;
-      if (!this.group) {
-        this.loading = false;
-        this.members = [];
-        return;
-      }
-      this.members = await this.$api.getMembers({groupId: this.group.id});
-      this.selected = [];
-      this.loading = false;
-    },
-    async removeMembers() {
-      const params = {
-        groupId: this.group.id,
-        userIds: this.selected
-      };
-      const response = await this.$api.removeMembers(params);
-      if (response === true) {
-        await this.getMembers();
-      }
-    }
+});
+
+const { users, loadUserData } = useAuthStore();
+
+const selected = ref([]);
+const toggleButton = new Button({
+  css: "btn-outline-primary",
+  icon: "person_add",
+  text: "Add members",
+});
+const modalButtons = [
+  new ModalButton({
+    css: "btn-primary",
+    icon: "person_add",
+    text: "Add",
+    callback: addMembers,
+  }),
+];
+
+async function onSelect() {
+  const elems = document.getElementsByName("select-users");
+  selected.value = Array.prototype.filter
+    .call(elems, (e) => e.checked)
+    .map((e) => e.value);
+}
+
+const members = ref([]);
+
+async function getMembers() {
+  if (!props.group) {
+    return
+  }
+  members.value = await api.getMembers({ groupId: props.group.id });
+  selected.value = [];
+}
+
+async function addMembers() {
+  const boxes = document.getElementsByName("add-users");
+  const addUserIds = Array.from(boxes)
+    .filter((input) => input.checked)
+    .map((input) => input.value);
+  const response = await api.addMembers({
+    groupId: props.group.id,
+    userIds: addUserIds,
+  });
+  if (response === true) {
+    await getMembers();
   }
 }
-</script>
 
-<style scoped>
+async function removeMembers() {
+  const params = {
+    groupId: props.group.id,
+    userIds: selected.value,
+  };
+  const response = await api.removeMembers(params);
+  if (response === true) {
+    await getMembers();
+  }
+}
 
-</style>
\ No newline at end of file
+await Promise.all([ loadUserData(), getMembers() ])
```

Comment:
Yeah, this is a bigger topic. I personally don't mind either, since the app already does a ton of api requests anyways. So to be safe I left this sort of behavior in place, but it sure can be fetched just once e.g. on app startup and then left as is and we'd let user do the page reload if they feel like there's something missing.

For a not frequently updated data (users, groups), it should be fairly harmless, right? I'll change this to a lazy-fetch instead.

Reply:
Cool, tanks :slightly_smiling_face:

Reply:
https://github.com/SUSE/aimaas/issues/200
https://gitlab.suse.de/tools/external-issues/-/issues/26